src/storage/src/hummock/compactor/compactor_runner.rs

```diff
@@ -842,11 +860,40 @@ where
             is_new_user_key = false;
         }
 
-        // Don't allow two SSTs to share same user key
-        sst_builder
-            .add_full_key(iter_key, value, is_new_user_key)
-            .verbose_instrument_await("add_full_key")
-            .await?;
+        // May drop stale columns
+        let check_table_id = last_key.user_key.table_id.table_id;
+        let mut is_value_rewritten = false;
+        if let HummockValue::Put(v) = value
+            && let Some(object_id) = object_id
+            && let Some(block_id) = block_id
+            && skip_schema_check.get(&object_id).map(|prev_block_id|*prev_block_id != block_id).unwrap_or(true)
+            && let Some(schema) = schemas.get(&check_table_id) {
```

Comment:
Is it possible that the schema can be stale? For instance, if a new column is added and written to the storage while the schema is still the old one, it may incorrectly identify the new column as invalid.

Reply:
Schema is at least as new as the one in storage, guaranteed by meta when issuing compaction task.